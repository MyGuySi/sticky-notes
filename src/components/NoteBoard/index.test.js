// @flow

import React from 'react';
import renderer from 'react-test-renderer';
import { render, screen, fireEvent } from '@testing-library/react';
import Board from './index'

const mockNotes = [
  { text: 'Test note', timestamp: 1590933321, color: 'pink' }
]

it('renders correctly', () => {
  const component = renderer.create(
    <Board
      title="Title"
      notes={mockNotes}
      onCreateNote={() => {}}
      onClickClearNotes={() => {}}
    />
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

it('switches to active state when add button is clicked', () => {
  render(
    <Board
      title="Title"
      notes={mockNotes}
      onCreateNote={() => {}}
      onClickClearNotes={() => {}}
    />
  )
  fireEvent.click(screen.getByTestId('add-button'));
  expect(screen.queryByTestId('add-button')).toBeNull();
  expect(screen.queryByTestId('text-input')).toBeTruthy();
  expect(screen.queryByTestId('save-button')).toBeTruthy();
  expect(screen.queryByTestId('clear-button')).toBeTruthy();
});

it('calls create handler when input has a value', () => {
  const handler = jest.fn();
  render(
    <Board
      title="Title"
      notes={mockNotes}
      onCreateNote={handler}
      onClickClearNotes={() => {}}
    />
  )
  fireEvent.click(screen.getByTestId('add-button'));
  fireEvent.change(screen.getByTestId('text-input'), { target: { value: 'test' }});
  fireEvent.click(screen.getByTestId('save-button'));
  expect(handler.mock.calls.length).toBe(1);
});

it('does not call create handler when input is empty', () => {
  const handler = jest.fn();
  render(
    <Board
      title="Title"
      notes={mockNotes}
      onCreateNote={handler}
      onClickClearNotes={() => {}}
    />
  )
  fireEvent.click(screen.getByTestId('add-button'));
  fireEvent.click(screen.getByTestId('save-button'));
  expect(handler.mock.calls.length).toBe(0);
});

it('calls clear handler', () => {
  const handler = jest.fn();
  render(
    <Board
      title="Title"
      notes={mockNotes}
      onCreateNote={() => {}}
      onClickClearNotes={handler}
    />
  )
  fireEvent.click(screen.getByTestId('add-button'));
  fireEvent.click(screen.getByTestId('clear-button'));
  expect(handler.mock.calls.length).toBe(1);
});