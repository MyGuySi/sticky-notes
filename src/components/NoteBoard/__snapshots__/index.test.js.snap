// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="stack"
>
  <h1
    className="heading"
  >
    Title
  </h1>
  <div
    className="grid"
  >
    <div
      className="wrapper"
    >
      <button
        className="button"
        data-testid="add-button"
        onClick={[Function]}
      >
        New note (n)
      </button>
    </div>
    <div
      className="note pink"
    >
      <div
        className="item"
      >
        <p
          className="message"
        >
          Test note
        </p>
        <p
          className="date"
        >
          50 years ago
        </p>
      </div>
    </div>
  </div>
  <button
    className="button"
    data-testid="clear-button"
    onClick={[Function]}
  >
    Clear all notes (c)
  </button>
</div>
`;
